EAF-D-2-2 library for edit docx

MrAmfix opened the pull request:
Created main class EDocx (src.edocx), that providing functions for adding comments, getting attributes (such as paraId) and editing the font style.

MrAmfix committed:
[feat]: added functions for adding comments and getting all attributes from paragraphs in EDocx-class

MrAmfix committed:
[feat]: added new functions
edit_font_size_by_id - edit font size
edit_color_by_id - edit text color
__para_ids - return dict with key: parad, value: object aragraph
and other small changes

MrAmfix committed:
[feat]: Added new functions and classes

slavamarcin commented on src/edocx.py:
```diff
@@ -0,0 +1,99 @@
+        """
+        :param path: path to DOCX document
+        """
+        if path[-5:] != '.docx':
```
Лучше сделать через split, вместо среза

slavamarcin commented on src/edocx.py:
```diff
@@ -0,0 +1,99 @@
+    @property
+    def __para_ids(self) -> Dict[str, Paragraph]:
+        res = {}
+        key = '{http://schemas.microsoft.com/office/word/2010/wordml}paraId'
```
внутри объекта Document python-docx есть переменная ns, которая представляет собой словарь схем docx, и по ключу ['w'] ты можешь получить нужную схему, т.е. key = f'{self.ns['w']}paraId', вроде как-то так

slavamarcin commented on src/edocx.py:
```diff
@@ -0,0 +1,99 @@
+            self.document.save(path)
+
+    @property
+    def __para_ids(self) -> Dict[str, Paragraph]:
```
Вопрос нужно ли делать через декоратор @property, т.к. у нас количество параграфов и их id вроде не собирается меняться, а в данном случае будет лишняя нагрузка и сократится скорость работы, т.к. каждый раз переменная __para_ids будет пересчитываться. (На ваше усмотрение)

slavamarcin commented on src/edocx.py:
```diff
@@ -0,0 +1,99 @@
+            para_ids[paraId].add_comment(comment, author)
+            if self.autosave:
+                self.save()
+            return True
```
Можно сделать обратно текущему, без else, таким образом сократиться количество строк и код будет выглядеть лаконичней

slavamarcin commented on src/edocx.py:
```diff
@@ -0,0 +1,99 @@
+            if bold is not None:
+                run.font.bold = bold
+            if underline is not None:
+                run.font.underline = underline
```
Сделай на каждый атрибут отдельные методы, а потом с помощью фабрики здесь используй

slavamarcin reviewed (commented):
Просмотрите мои комментарии и примите во внимание.

MrAmfix committed:
[feat/refactor]: Split edit_style function and added new error class
edit_style_by_id function has been splitted into sub-functions, added new error-class and other small changes

Vl-Tershch commented on src/exceptions.py:
```diff
@@ -0,0 +1,17 @@
+class ParagraphNotFound(Exception):
+    """
+    Special class
+    """
```
Точно стоит добавить описание этим исключениям, а не Special class везде. И, возможно, стоит внутри писать реализацию, как например обсуждают [здесь](https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python).

Vl-Tershch commented on src/utils.py:
Мне кажется, что следует разделить эти классы на отдельные файлы, и вложить их, например, в директорию utils.

Vl-Tershch reviewed (commented):
Отсутствуют модульные тесты к реализованному функционалу, нужно добавить. Также добавь пожалуйста в описание PR'a что было сделано.

MrAmfix committed:
[tests/refactor] Created folder src.enum, added unit-test for exceptions

MrAmfix committed:
[refactor]: moved files in their new repository according with new structure of project and rename some classes.

MrAmfix committed:
[fix]: The path to the module in the docxrefactor.py was changed (it was specified incorrectly)

MrAmfix committed:
Merge branch 'development' of https://github.com/LISA-ITMO/Python-Documents-Modifier into EAF-D-2--2-library-for-edit-DOCX

MrAmfix committed:
[feat]: added requirement.txt

MrAmfix committed:
[fix]: fixed build-test.yml

MrAmfix committed:
[fix]: added full paths to files in tests

MrAmfix committed:
[fix]: found paths to files with os.path

MrAmfix committed:
[refactor]: created exceptions folder in src, renamed docxredactor.py and exceptions_docx.py, moved docx_exceptions.py in src/exceptions/
